zerver/lib/actions.py

```diff
@@ -5036,6 +5036,52 @@ def do_mark_stream_messages_as_read(
     return count
 
 
+def do_mark_user_messages_as_read(
+    user_profile: UserProfile,
+    muted_user_profile: UserProfile,
+) -> int:
+    msgs = UserMessage.objects.filter(
+        user_profile=user_profile,
+    )
+
+    msgs = msgs.filter(message__sender=muted_user_profile)
+
+    msgs = msgs.extra(
+        where=[UserMessage.where_unread()],
+    )
```

Comment:
I'd prefer that these first several lines were a single block rather than reassigning the variable several times; also, we should call the variable `messages` -- our naming convention is to avoid abbreviations like `msgs`.